### PR TITLE
fix(test): data race in TestEventsChronologicalSorter_Start

### PR DESCRIPTION
**Pull request by danielpacak**
Resolves: #1630

Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: pkg/events/sorting/sorting_test.go [hunk 1]
```diff
@@ -297,13 +299,19 @@ func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan chan *t
...
 }
 
-func sendTestEvents(sorterInputChannel chan *trace.Event, eventPool []eventsIteration) {
-	startTime := int(time.Now().UnixNano())
-	for _, eventsIteration := range eventPool {
-		time.Sleep(eventsIteration.delay)
-		for _, event := range eventsIteration.events {
-			event.Timestamp = startTime + (event.Timestamp * int(time.Millisecond.Nanoseconds()))
-			sorterInputChannel <- &event
+func sendTestEvents(eventPool []eventsIteration) <-chan *trace.Event {
```
Comment (danielpacak) [inline pkg/events/sorting/sorting_test.go:302]: Make sendTestEvents the owner of the events channel. This is a common pattern to reason about Go concurrent programs. 
Beyond that, close the channel explicitly and use non buffered channel it's easier to catch data races.

File: pkg/events/sorting/sorting_test.go [hunk 2]
```diff
@@ -273,13 +272,16 @@ func TestEventsChronologicalSorter_Start(t *testing.T) {
...
 
-func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan chan *trace.Event, fatalErrorsChan chan error) ([]trace.Event, error) {
+func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan <-chan *trace.Event, fatalErrorsChan chan error) ([]trace.Event, error) {
 	ticker := time.NewTicker(time.Second)
 	outputList := make([]trace.Event, 0)
 	eventsReceived := 0
 	for {
 		select {
-		case event := <-sorterOutputChan:
+		case event, ok := <-sorterOutputChan:
```
Comment (danielpacak) [inline pkg/events/sorting/sorting_test.go:281]: I think in the sorter itself we're not checking for the closed channel, but it's not relevant to fix the DATA RACE warnings.

Comment (AlonZivony) [reply pkg/events/sorting/sorting_test.go:281]: I will check it

Comment (rafaeldtinoco) [reply pkg/events/sorting/sorting_test.go:281]: @AlonZivony any feedback here ?

**Pull request merged**